components/tidb_query/src/expr/builtin_json.rs

```diff
@@ -204,6 +218,115 @@ mod tests {
     use crate::expr::{EvalContext, Expression};
     use tipb::ScalarFuncSig;
 
+    #[test]
+    fn test_json_length() {
+        let cases = vec![
+            (None, None, None),
+            (Some("null"), None, None),
+            (Some(r#"{"a":{"a":1},"b":2}"#), Some(b"$".to_vec()), Some(2)),
+            (Some("1"), None, Some(1)),
+            (
+                Some(r#"{"a": [1, 2, {"aa": "xx"}]}"#),
+                Some(b"$.*".to_vec()),
+                None,
+            ),
+            (Some(r#"{"a":{"a":1},"b":2}"#), Some(b"$".to_vec()), Some(2)),
+            // Tests with path expression
+            (
+                Some(r#"[1,2,[1,[5,[3]]]]"#),
+                Some(b"$[2]".to_vec()),
+                Some(2),
+            ),
+            (Some(r#"[{"a":1}]"#), Some(b"$".to_vec()), Some(1)),
+            (
+                Some(r#"[{"a":1,"b":2}]"#),
+                Some(b"$[0].a".to_vec()),
+                Some(1),
+            ),
+            (Some(r#"{"a":{"a":1},"b":2}"#), Some(b"$".to_vec()), Some(2)),
+            (
+                Some(r#"{"a":{"a":1},"b":2}"#),
+                Some(b"$.a".to_vec()),
+                Some(1),
+            ),
+            (
+                Some(r#"{"a":{"a":1},"b":2}"#),
+                Some(b"$.a.a".to_vec()),
+                Some(1),
+            ),
+            (
+                Some(r#"{"a": [1, 2, {"aa": "xx"}]}"#),
+                Some(b"$.a[2].aa".to_vec()),
+                Some(1),
+            ),
+            // Tests without path expression
+            (Some(r#"{}"#), None, Some(0)),
```

Comment:
for this case,  the length of args is 2 and the second argument is Null, we should return null;
```
mysql> select json_length("[1,2,[1,3]]");
+----------------------------+
| json_length("[1,2,[1,3]]") |
+----------------------------+
|                          3 |
+----------------------------+
1 row in set (0.00 sec)

mysql> select json_length("[1,2,[1,3]]", NULL);
+----------------------------------+
| json_length("[1,2,[1,3]]", NULL) |
+----------------------------------+
|                             NULL |
+----------------------------------+
1 row in set (0.00 sec)
```

Reply:
Done